app/controllers/posts_controller.rb

```diff
@@ -7,5 +8,28 @@ def index
   def show
     @post = Post.includes(comments: :author).find(params[:id])
     @user = @post.author
+    @new_like = Like.new
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params)
+    @user = User.find(params[:id])
+
+    if @post.save
+      redirect_to user_post_path(@user, @post), notice: 'Post successfully created'
+    else
+      redirect_to new_user_post_path(@user), alert: 'Failed to create post'
+    end
+  end
+
+  private
+
```

Comment:
- [ ] Great job fixing the error however kindly ensure that creating a Post is possible. Currently this is not the case. And since I am unable to create a post I cannot confirm that a user can add comments and likes to a post, kindly ensure this too is possible.


ScreenCast
[cinnamon-2024-01-20T232830+0200.webm](https://github.com/ReemMohamedAbdelfatah/rails-blog-app/assets/81932478/7a2f1caa-5f77-4fc4-b059-0db7b70a4291)